spec/models/download_spec.rb

```diff
@@ -96,39 +96,44 @@
     end
   end
 
-  describe "authors" do
-    let(:work) { Work.new }
+  describe "page_title" do
+    let(:fandom1) { create(:canonical_fandom) }
```

Comment:
Most of these tests don't actually require works, pseuds, fandoms, etc. to be saved to the database, so you should be able to write them using [`build`](https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#build-strategies) and make them faster, e.g.

https://github.com/otwcode/otwarchive/blob/14fbeb6ae13f9637cfdd3c4f586620d8863ff32a/spec/models/download_spec.rb#L60-L66

Otherwise this looks good to me.

Reply:
Thanks for the comment, and that's true. I've managed to use ``build`` in most of them, but not  ``let(:work)``. (To be honest, I only understand the Work class on a very shallow level, and I'm not sure which part of it around saving/validation is the issue.)

Reply:
I figured this out. If we use `work.fandoms.string`, we call this method:

https://github.com/otwcode/otwarchive/blob/015ebc1d01be7c3962d59082b5f9515ec6e7c599/app/models/tag.rb#L459-L463

which will force a database query.

You can switch from `work.fandoms.string` to `work.fandom_string` instead, which will use `.fandoms` on an unsaved work without hitting the database:

https://github.com/otwcode/otwarchive/blob/015ebc1d01be7c3962d59082b5f9515ec6e7c599/app/models/concerns/taggable.rb#L27-L29

Then you should be able to avoid saving the work:
```rb
let(:work) { build(:work, fandoms: [fandom1, fandom2], title: "Foo bar") }
```

Reply:
Thanks again! This has been very illuminating. I'll revise it again (hopefully for the last time).